lib/_http_server.js

```diff
@@ -996,13 +997,18 @@ function parserOnIncoming(server, socket, state, req, keepAlive) {
   res.shouldKeepAlive = keepAlive;
   res[kUniqueHeaders] = server[kUniqueHeaders];
 
-  if (onRequestStartChannel.hasSubscribers) {
-    onRequestStartChannel.publish({
+  const dcMessage = (
+    onRequestStartChannel.hasSubscribers ||
+    onRequestEndChannel.hasSubscribers
+  ) ? {
       request: req,
       response: res,
       socket,
       server
-    });
+    } : undefined;
+
+  if (onRequestStartChannel.hasSubscribers) {
```

Comment:
I assume this needs to be changed now to call `AsyncLocalStore.runInTracingChannel`